src/connectors/google.rs

```diff
@@ -12,18 +12,89 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use gouth::Token;
 use std::sync::Arc;
 use tonic::metadata::MetadataValue;
 use tonic::service::Interceptor;
 use tonic::{Request, Status};
 
-pub(crate) struct AuthInterceptor {
-    pub token: Box<dyn Fn() -> ::std::result::Result<Arc<String>, Status> + Send>,
+pub trait TokenProvider: Clone {
```

Comment:
let's avoid pubing things if we can :)
```suggestion
pub(crate) trait TokenProvider: Clone {
```